app/lib/actions/recurring_action.rb

```diff
@@ -0,0 +1,27 @@
+module Actions
+  module RecurringAction
+    # When included sets the base action to use the RecurringLogic middleware and configures
+    #   #trigger_repeat to be called when appropriate to trigger the next repeat.
+    def self.included(base)
+      base.middleware.use Actions::Middleware::RecurringLogic
+      base.execution_plan_hooks.use :trigger_repeat, :on => [:planned, :failure]
+    end
+
+    # Hook to be called when a repetition needs to be triggered. This either happens when the plan goes into planned state
+    #   or when it fails.
+    def trigger_repeat(kind, execution_plan)
+      if (kind == :planned || execution_plan.result == :error) && execution_plan.delay_record && recurring_logic_task_group
```

Comment:
What is the situation that `(kind == :planned || execution_plan.result == :error)` would not be true?

Reply:
Right, now that I'm thinking about it it should not happen, unless someone explicitly adds another `execution_plan_hooks.use :trigger_repeat` line to the action